src/main/kotlin/com/wgslfuzz/core/ShaderJob.kt

```diff
@@ -210,6 +212,8 @@ private fun literalExprFromBytes(
     bufferByteIndex: Int,
 ): Pair<Expression, Int> {
     assert(bufferByteIndex % 4 == 0)
+    // The resolved type may be a reference. We only care about the store type of the reference, so remove the outer reference.
```

Comment:
We shouldn't expect to get a reference type here. This is because we are trying to turn the bytes supplied for a uniform value into a vector expression. The uniform value will either be a basic type like `u32`, or a vector or matrix, or a `struct` with fields that are basic types, vectors / matrices or nested structs. So it should be safe to remove this, and if I'm wrong an a reference comes along we'll find out thanks to the `else` below.

Reply:
Although for some reason, the struct test cause does not appear to have reference type and so passes. I'll see if I can figure out why.

Reply:
Maybe the right solution would be here;

```
        val uniformType: Type = getUniformType(environment, tu.getUniformDeclaration(group, binding))
```

... to apply the dereferencing here (or perhaps even in `getUniformType` - maybe it could be renamed `getUniformStoreType`)?

Reply:
It's because of the following:
```
private fun getUniformType(
    environment: ResolvedEnvironment,
    uniformDeclaration: GlobalDecl.Variable,
) = if (uniformDeclaration.typeDecl is TypeDecl.NamedType) {
    getUniformTypeByName(environment, uniformDeclaration.typeDecl.name)
} else {
    getUniformTypeByName(environment, uniformDeclaration.name)
}
```
A struct is the first case and so `getUniformTypeByName` looks up the named type of the struct and returns the naked type of the struct without a reference. Other uniform buffers such as ints will take the second branch and look up the type of the variable which will be wrapped in a reference.